
<launch>

   <!-- ROBOT MAPPING VERSION: use this with ROS bag demo_mapping.bag -->
   <!-- WARNING : Database is automatically deleted on each startup -->
   <!--           See "delete_db_on_start" option below... -->

  <param name="use_sim_time" type="bool" value="True"/>

  <!-- SLAM (robot side) -->
  <!-- args: "delete_db_on_start" and "udebug" -->
  <node name="rtabmap" pkg="rtabmap" type="rtabmap" output="screen" args="--delete_db_on_start">
	<param name="frame_id" type="string" value="base_footprint"/>
	
	<param name="subscribe_depth" type="bool" value="true"/>
	<param name="subscribe_laserScan" type="bool" value="true"/>
	
	<remap from="odom" to="/az3/base_controller/odom"/>
	<remap from="scan" to="/jn0/base_scan"/>

	<remap from="rgb/image" to="data_throttled_image"/>
  	<remap from="depth/image" to="data_throttled_image_depth"/>
  	<remap from="rgb/camera_info" to="data_throttled_camera_info"/>
  	
  	<param name="rgb/image_transport" type="string" value="compressed"/>
    <param name="depth/image_transport" type="string" value="compressedDepth"/>
	
	<param name="queue_size" type="int" value="10"/>
	
	<!-- RTAB-Map's parameters: do "rosrun rtabmap rtabmap (double-dash)params" to see the list of available parameters. -->
	<param name="RGBD/ScanMatchingSize" type="string" value="1"/>  <!-- Do odometry correction with consecutive laser scans -->
	<param name="RGBD/LocalLoopDetectionSpace" type="string" value="true"/>  <!-- Laser scan matching when near (using estimated position) old locations in WM -->
	<param name="RGBD/LocalLoopDetectionTime" type="string" value="false"/>  <!-- Don't find loop closures in time (i.e. with locations in STM) -->
  </node>
  
  <!-- Visualisation (client side) -->
  <node pkg="rtabmap" type="rtabmapviz" name="rtabmapviz" args="-d $(find rtabmap)/launch/config/rgbd_gui.ini" output="screen">
  	  <param name="subscribe_depth" type="bool" value="true"/>
      <param name="subscribe_laserScan" type="bool" value="true"/>
      <param name="queue_size" type="int" value="10"/>
      <param name="frame_id" type="string" value="base_footprint"/>
    
      <remap from="rgb/image" to="data_throttled_image"/>
      <remap from="depth/image" to="data_throttled_image_depth"/>
      <remap from="rgb/camera_info" to="data_throttled_camera_info"/>
      <remap from="scan" to="/jn0/base_scan"/>
      <remap from="odom" to="/az3/base_controller/odom"/>
      
      <param name="rgb/image_transport" type="string" value="compressed"/>
      <param name="depth/image_transport" type="string" value="compressedDepth"/>
  </node>

</launch>
